package tests.validation.names.duplicates.inCallableType

segment s(
    f: (
        // $TEST$ no error r"A parameter with name '\w*' exists already\."
        »duplicateParameter«: Int,
        // $TEST$ error "A parameter with name 'duplicateParameter' exists already."
        »duplicateParameter«: Int,
        // $TEST$ no error r"A parameter with name '\w*' exists already\."
        »uniqueParameter«: Int,
        // $TEST$ no error r"A parameter with name '\w*' exists already\."
        »parameterAndResult«: Int
    ) -> (
        // $TEST$ no error r"A result with name '\w*' exists already\."
        »duplicateResult«: Int,
        // $TEST$ error "A result with name 'duplicateResult' exists already."
        »duplicateResult«: Int,
        // $TEST$ no error r"A result with name '\w*' exists already\."
        »uniqueResult«: Int,
        // $TEST$ no error r"A result with name '\w*' exists already\."
        »parameterAndResult«: Int
    )
) {}
